### PR TITLE
Add helm charts

### PR DESCRIPTION
**Pull request by rmstoi**
Signed-off-by: Ruslan Mstoi <ruslan.mstoi@intel.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -232,3 +232,7 @@ _work/venv/.stamp: docs/requirements.txt
 	python3 -m venv ${@D}
 	. ${@D}/bin/activate && pip install wheel && pip install -r $<
 	touch $@
+
+helm:
+	cp -rf deployments/operator/* charts/operator
```
Comment (mythi) [inline Makefile:237]: @rmstoi did you try the `make generate` approach?

Comment (mythi) [inline Makefile:237]: @rmstoi did you try the `make generate` approach?

Comment (mythi) [reply Makefile:237]:
```
diff --git a/Makefile b/Makefile
index 4b1f81b1..648d5455 100644
--- a/Makefile
+++ b/Makefile
@@ -95,6 +95,9 @@ generate:
        $(CONTROLLER_GEN) crd:crdVersions=v1 \
                paths="./pkg/apis/..." \
                output:crd:artifacts:config=deployments/operator/crd/bases
+       $(CONTROLLER_GEN) crd:crdVersions=v1 \
+               paths="./pkg/apis/..." \
+               output:crd:artifacts:config=charts/operator/crds
        $(CONTROLLER_GEN) crd:crdVersions=v1 \
                paths="./pkg/apis/fpga/..." \
                output:crd:artifacts:config=deployments/fpga_admissionwebhook/crd/bases
$ make generate
$ ls charts/operator/crds
```

Comment (rmstoi) [reply Makefile]: @mythi great idea with makefile change. I pushed updated PR.

File: charts/operator/values.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+image:
+  hub: intel
+  name: intel-device-plugins-operator
+  tag: "latest"
```
Comment (mythi) [inline charts/operator/values.yaml:4]:
```suggestion
  tag: "devel"
```

File: charts/operator/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v2
+name: intel-device-plugins-operator
+description: A Helm chart for Intel Device Plugins Operator for Kubernetes
+type: application
+version: 0.1.0
+appVersion: "0.23.0"
```
Comment (mythi) [inline charts/operator/Chart.yaml:6]: we have `make set-version` helper that helps to manage the release versions in various places. Can you check the Helm related components get updated properly too?

**Comment by bart0sh** [comment]
@rmstoi It would be nice to have a documentation explaining what is it for and how to use it.

File: scripts/set-version.sh [hunk 4]
```diff
@@ -15,6 +15,6 @@ if [ $# != 1 ] || [ "$1" = "?" ] || [ "$1" = "--help" ]; then
     exit 1
 fi
 
-for file in $(git grep -l '^TAG?*=\|intel/crypto-perf:\|intel/opae-nlb-demo:\|intel/intel-[^ ]*:\|version=' Makefile deployments demo/*fpga*.yaml pkg/controllers/*/*_test.go build/docker/*.Dockerfile); do
-    sed -i -e "s;\(^TAG?*=\|intel/crypto-perf:\|intel/opae-nlb-demo:\|intel/intel-[^ ]*:\|version=\)[^ \"]*;\1$1;g" "$file";
+for file in $(git grep -l '^TAG?*=\|intel/crypto-perf:\|intel/opae-nlb-demo:\|intel/intel-[^ ]*:\|version=\|appVersion:' Makefile deployments demo/*fpga*.yaml pkg/controllers/*/*_test.go build/docker/*.Dockerfile charts); do
```
[the remark this answers is not in the document]
Comment (rmstoi) [reply scripts/set-version.sh]: I've made a commit to update tag in values.yaml to the value of TAG variable. Hence, in charts there are now appVersion and tag set to the value of TAG. It this ok?

**Comment by rmstoi** [comment]
> @rmstoi It would be nice to have a documentation explaining what is it for and how to use it.

@bart0sh To which file should the documentation be added?

File: scripts/set-version.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (mythi) [reply scripts/set-version.sh]: @rmstoi LGTM

**Comment by bart0sh** [comment]
> To which file should the documentation be added?

charts/operator/README.md would be a good place + it should be reffered from the top level README.md.

**Comment by rmstoi** [comment]
> > To which file should the documentation be added?
> 
> charts/operator/README.md would be a good place + it should be reffered from the top level README.md.

I've added preliminary documentation, which could grow as the helm support goes further.
I put the README.md to charts (not charts/operator) to exclude it from inclusion in the chart archive files.

**Review by mythi — APPROVED** [review]

**Pull request merged**